### DISCUSSION TITLE
Nearest point 3D optimized (taking into account the height difference)

### DISCUSSION THREAD

**Discussion by heu3becteh** [dtcenter/METplus#2186] [Configuration] [answered]
Hello!

So as I understand right now `topo_mask` is used by **Point-Stat** only to filter the obs/fcs data not satisfying thresholds `use_obs_thresh` (discarding the stations with height difference above the threshold) and `interp_fcst_thresh` (discarding the forecast grid points with height difference above the threshold).

**Geography Match** uses these thresholds and `land_mask` to filter land/water difference between observation and forecast.

Is there any other way to use height difference between observation site and forecast grid point?
I would like to use the height difference to find Nearest Neighbor: `min(horizontal_distance + vertical_distance*100)`. Is it possible?
May it be possible to use height difference as weights or some other way, apart from discarding the data not satisfying the threshold?

Best regards,
Alexander

**Comment by j-opatz** [comment, marked as answer]
Hi Alexander,

And thank you for your question. Unfortunately, there's no way currently in MET to use the height difference in the manner you're describing. As you noted, `topo_mask` allows filtering of observations and forecasts compared to an intermediate height difference of the topo map. This step is the closest you can get to comparing your forecast to observation height differences, as `topo_mask` will assume the topo map definition is within the provided forecast file if the `file_name` entry isn't used.

But from your description, it sounds like you'd like to extract the actual height distances to be used for nearest neighbor and weighting. Nearest Neighbor does exist as a [regridding method](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#regrid) in `regrid`, but those values cannot be accessed or saved in any method.

**Reply by heu3becteh** [reply]
Thank you for the reply!
> But from your description, it sounds like you'd like to extract the actual height distances to be used for nearest neighbor and weighting. Nearest Neighbor does exist as a [regridding method](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#regrid) in `regrid`, but those values cannot be accessed or saved in any method.

Yes, I want to use height difference together with horizontal distance when looking for the nearest neighbour.
I will try looking into possibilities to make it work like that, but probably will just use `topo_mask` filtering in the end.
